### PR TITLE
feat: Enable setting mandatory and persistent in upstream healthchecks in VS and VSR

### PR DESCRIPTION
**Pull request by ciarams87**
### Proposed changes
See [mandatory-health-checks](https://docs.nginx.com/nginx/admin-guide/load-balancer/http-health-check/#mandatory-health-checks)

The mandatory parameter requires every newly added server to pass all configured health checks before NGINX Plus sends traffic to it. When combined with slow start, it gives a new server more time to connect to databases and “warm up” before being asked to handle their full share of traffic. Mandatory health checks can be marked as persistent, so that the previous state is remembered when reloading configuration. 

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by pleshakov — APPROVED** [review]
🚀  (with some doc related suggestions)

File: docs/content/configuration/virtualserver-and-virtualserverroute-resources.md [hunk 1]
```diff
@@ -369,6 +372,8 @@ Note: This feature is supported only in NGINX Plus.
...
 |``grpcStatus`` | The expected [gRPC status code](https://github.com/grpc/grpc/blob/master/doc/statuscodes.md#status-codes-and-their-use-in-grpc) of the upstream server response to the [Check method](https://github.com/grpc/grpc/blob/master/doc/health-checking.md). Configure this field only if your gRPC services do not implement the gRPC health checking protocol. For example, configure ``12`` if the upstream server responds with `12 (UNIMPLEMENTED)` status code. Only valid on gRPC type upstreams. | ``int`` | No |
 |``grpcService`` | The gRPC service to be monitored on the upstream server. Only valid on gRPC type upstreams. | ``string`` | No |
+|``mandatory`` | Require every newly added server to pass all configured health checks before NGINX Plus sends traffic to it. If this is not specified, or is set to false, the server will be initially considered healthy. When combined with [slow_start](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#slow_start), it gives a new server more time to connect to databases and “warm up” before being asked to handle their full share of traffic. | ``bool`` | No |
```
[the remark this answers is not in the document]
Comment (ciarams87) [reply docs/content/configuration/virtualserver-and-virtualserverroute-resources.md]: I presume removing the ` | ``bool`` | No |` was unintentional, so I made this change locally. Thanks for catching it!

File: docs/content/configuration/virtualserver-and-virtualserverroute-resources.md [hunk 2]
```diff
@@ -324,12 +324,13 @@ Note: This feature is supported only in NGINX Plus.
 
 ### Upstream.Healthcheck
 
-The Healthcheck defines an [active health check](https://docs.nginx.com/nginx/admin-guide/load-balancer/http-health-check/). In the example below we enable a health check for an upstream and configure all the available parameters:
+The Healthcheck defines an [active health check](https://docs.nginx.com/nginx/admin-guide/load-balancer/http-health-check/). In the example below we enable a health check for an upstream and configure all the available parameters, including combining with the `slow-start` directive for [mandatory healthchecks](https://docs.nginx.com/nginx/admin-guide/load-balancer/http-health-check/#mandatory-health-checks):
```
[the remark this answers is not in the document]
Comment (ciarams87) [reply docs/content/configuration/virtualserver-and-virtualserverroute-resources.md]: Done, thanks!

**Review by soneillf5 — APPROVED** [review]

**Comment by soneillf5** [comment]
@ciarams87 sorry for the late review

**Pull request merged**